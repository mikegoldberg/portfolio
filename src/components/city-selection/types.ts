export interface CitySelectionProps {
  city: string;
  onSubmit: (name: string) => void;
}
