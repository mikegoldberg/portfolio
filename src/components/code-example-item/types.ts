import { MouseEventHandler } from "react";

export interface CodeExampleItemProps {
  title: string;
  poster: string;
  onClick: MouseEventHandler;
  source: string;
}
