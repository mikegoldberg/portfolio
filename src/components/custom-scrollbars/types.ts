import { ReactElement } from "react";

export interface CustomScrollbarsProps {
  onScrollFrame: (position: number) => void;
  children?: ReactElement;
}
