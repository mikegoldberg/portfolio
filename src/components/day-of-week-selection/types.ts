type OnDaySelected = {
  (date: string): void;
};

export interface DayOfWeekSelectionProps {
  dates: Array<string>;
  activeDayOfWeek: string;
  onDaySelected: OnDaySelected;
}
