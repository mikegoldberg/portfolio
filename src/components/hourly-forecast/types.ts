export type HourlyForecastItemProps = {
  startTime: string;
  temperature: string;
  temperatureUnit: string;
  windDirection: string;
  temperatureChange: number;
  windSpeedValue: string;
  windSpeedUnit: string;
  iconLarge: string;
  shortForecast: string;
};

export interface HourlyForecastProps {
  items: Array<HourlyForecastItemProps> | unknown[];
  onScrollFrame: (position: number) => void;
}

export interface HourlyForecastHandler {
  scrollToDate: (date: string) => void;
  getDayOfWeekAtItemIdx: (idx: number) => string;
}

export interface TemperatureProps {
  temp: string;
  unit: string;
}

export interface WindDirectionProps {
  windDirection: string;
  windSpeed: string;
  windSpeedUnit: string;
}
