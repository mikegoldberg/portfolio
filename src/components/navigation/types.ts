export interface MobileHeadingProps {
  onOpen: () => void;
}

export interface NavMenuProps {
  onNavClicked?: () => void;
}

export interface NavLinkProps {
  label: string;
  to: string;
  onNavClicked?: () => void;
}
