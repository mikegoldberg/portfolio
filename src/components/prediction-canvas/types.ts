export type PredictionCanvasProps = {
  hide: boolean;
  width: number;
  height: number;
};
