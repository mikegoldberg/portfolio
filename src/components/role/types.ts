type Title = string;
type Company = string;
type Description = string;
type Skillset = Array<string>;
type Achievements = Array<string>;
type Start = string;
type End = string | null;
type Contract = boolean;

export interface RoleProps {
  title: Title;
  company: Company;
  description: Description;
  skillset: Skillset;
  achievements: Achievements;
  start: Start;
  end: End;
  contract: Contract;
}
