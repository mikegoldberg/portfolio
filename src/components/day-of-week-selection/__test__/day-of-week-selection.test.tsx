import { render } from "../../../test-utils";
import DayOfWeekSelection from "..";
import { DayOfWeekSelectionProps } from "../types";

const defaultProps: DayOfWeekSelectionProps = {
  dates: ["12-01-2022"],
  activeDayOfWeek: "12-01-2022",
  onDaySelected: () => {},
};

describe("DayOfWeekSelection", () => {
  it("renders correctly", () => {
    const { container } = render(<DayOfWeekSelection {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
