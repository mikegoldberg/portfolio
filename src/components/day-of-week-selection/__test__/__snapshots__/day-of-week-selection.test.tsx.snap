// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayOfWeekSelection renders correctly 1`] = `
<div>
  <div
    class="css-qhu0uq"
  >
    <button
      class="chakra-button css-1oirpja"
      data-date="12-01-2022"
      data-testid="day-of-week-selection-day-of-week"
      type="button"
    >
      Thu
    </button>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
